6211 Карпинская Анастасия Михайловна Лаб.4 Вар.8

stasya-kk opened the pull request:
ФИО: Карпинская Анастасия  
Номер группы: 6211  
Номер лабораторной: 4
Номер варианта: 8  
Краткое описание решаемой задачи: Согласно варианту 8 лабораторной 4 мне надо было добавить столбцы со значениями для гистограмм распределения яркости по каждому каналу (r, g, b) изображения. На графике отобразить 3 гистограммы.
Краткое описание работы кода: Модуль dataframe_manager.py отвечает за работу с DataFrame - создание из аннотации, добавление колонок с данными о яркости, сортировку и фильтрацию по диапазонам яркости. Модуль image_analyzer.py анализирует изображения, вычисляет гистограммы яркости по каналам R, G, B и определяет диапазоны яркости для каждого изображения. Модуль visualizer.py создает визуализации - гистограммы распределения яркости по каналам и общее распределение файлов по диапазонам яркости.
Модуль main.py является точкой входа в приложение. Чтобы его запустить, требуется указать аргументы командной строки:
python main.py --annotation annotations.csv --output_df result.csv --output_plot distribution.png

Commit:
сделала лабу 1

Commit:
Исправила ошибки, также теперь для запуска теперь нужно вводить python main.py data.txt moscow_people.txt

stasya-kk committed:
Merge branch 'itsecd:main' into main

stasya-kk committed:
сделана лаба 4

stasya-kk committed:
Delete moscow_people.txt

microalan commented on dataframe_manager.py:
```diff
@@ -0,0 +1,127 @@
+        """Сортирует DataFrame по диапазону яркости."""
+        try:
+            # Создаем порядок сортировки для диапазонов
+            range_order = {"0-85": 0, "86-170": 1, "171-255": 2}
```
Добавить возможность пользователю задавать диапазоны

microalan reviewed (commented):
Убрать print из всех функций, кроме main и функции печати

stasya-kk committed:
внесла изменения

также теперь при запуске необходимо указывать  python main.py --annotation annotations.csv --output_df output.csv --output_plot plot.png --ranges "0-50,51-150,151-255"(при необходимости менять значения диапазонов)

stasya-kk committed:
Delete result.csv

stasya-kk committed:
Delete distribution.png

microalan reviewed (approved):
Зачтено
